datafusion/datasource-parquet/src/opener.rs

```diff
@@ -176,6 +187,17 @@ impl FileOpener for ParquetOpener {
             // We'll also check this after every record batch we read,
             // and if at some point we are able to prove we can prune the file using just the file level statistics
             // we can end the stream early.
+            if !constant_columns.is_empty() {
+                predicate = predicate
+                    .map(|expr| {
+                        if is_dynamic_physical_expr(&expr) {
```

Comment:
Why do we need this clause? What breaks if we remove it? I'd think that rewriting the dynamic expression would work - it would try to rewrite it's children, which shouldn't cause any issues. Once `snapshot` is called the produces expression should have the remapped children.